### PR TITLE
001-notifications.30s.py: fix small bugs and add discussion

### PR DESCRIPTION
**Pull request by elalemanyo**
- Fix /issues/1780
- Fix `TypeError: a bytes-like object is required, not 'str'`
- Add `Discussion`, not sure how to add the image 🙈 I wanted to use this one: https://primer.style/octicons/comment-discussion-16
- Add also `CheckSuite`, same problem here with the image. https://primer.style/octicons/play-16, not sure why but `CheckSuite` don't have `href` so link is not working 😞

### CODE REVIEW

File: Dev/GitHub/notifications.30s.py [hunk 1]
```diff
@@ -185,6 +185,11 @@ def format_notification(notification):
             "image"
         ] += "JdJREFUKJGl0DsKwkAUBdDTRgvFHbgmNyLY+QWzKxM/kK2kSKc70MIIQ0ziqBceA/dxinn8mSkKVMGUmH+CBWaNboQjdn2wqt97Pa8kNd5+C0O86YNdSZC34RLjCJxhHZYLXDCIxKuwTHGOwBNcm2WKUw9OcMCybZl6XjHpQOs30cB5gKNQiDPPP0WjV/a4aVwxNsNfUGce7P8k4XgVPSYAAAAASUVORK5CYII="
         formatted["templateImage"] = formatted.pop("image")
+    elif type == "Discussion":
+        formatted[
+            "image"
+        ] += ""
+        formatted["templateImage"] = formatted.pop("image")
```
Comment (elalemanyo) [inline Dev/GitHub/notifications.30s.py:192]:
```suggestion
    elif type == "Discussion":
        formatted[
            "image"
        ] += ""
        formatted["templateImage"] = formatted.pop("image")
    elif type == "CheckSuite":
        formatted[
            "image"
        ] += ""
        formatted["templateImage"] = formatted.pop("image")
```

**Comment by matryer** [comment]
@elalemanyo what's happening with the option click?

**Comment by elalemanyo** [comment]
> @elalemanyo what's happening with the option click?

@matryer `mark as read` is not working, notification still there 🤷 not sure what is going on

**Comment by gingerbeardman** [comment]
Can confirm this version fixes issues with the python3 "fixed" version that is the current download. But "mark as read" function is not working on option click.

Personally, I just reinstalled Python2 and continue to use my old version.

**Comment by elalemanyo** [comment]
> Can confirm this version fixes issues with the python3 "fixed" version that is the current download. But "mark as read" function is not working on option click.
> 
> Personally, I just reinstalled Python2 and continue to use my old version.

you are right, using the last python2 version works like a charm

**Comment by gingerbeardman** [comment]
The only thing the last Python 2 version does not cope with are Discussion notifications.

**Comment by elalemanyo** [comment]
Yes `Discussion` and `CheckSuite` 😞 Not sure what is the problem with "mark as read"  function with the python3 version...

**Comment by gingerbeardman** [comment]
Sadly i do not have time to look into that, I will try to help when i can.

**Comment by gingerbeardman** [comment]
"mark as read" problem is because `data` variable is not correct type for `urlopen`, I added debugging and get this error message. This is a typical problem when porting from Python 2 to 3.

> TypeError: POST data should be bytes, an iterable of bytes, or a file object. It cannot be of type str.

I will investigate the solution.

**Comment by funkymonkeymonk** [comment]
Hey. I found this open PR looking for the process to upgrade the script to python3. I was wondering what work is remaining before this is good to merge? I have some time and would be happy to help carry this over the finish line instead of starting from scratch.

**Pull request merged**